test/e2e/service.go

```diff
@@ -1090,8 +1117,11 @@ spec:
 
 		ginkgo.By("by sending a TCP packet to service " + svcName + " with type=LoadBalancer in namespace " + namespaceName + " from backend pod " + backendName)
 
-		_, err = curlInContainer(clientContainer, svcIP, endpointHTTPPort, "big.iso -o big.iso", 120)
-		framework.ExpectNoError(err, "failed to curl load balancer service")
+		for i := 0; i < 5; i++ {
```

Comment:
@trozet : does this work? So instead of shifting the backends across two nodes I am curling the service 5 times so that should increase the chance of it hitting a different node at least once.

Reply:
I think it is an OK strategy, but we need to confirm that the both paths are actually tested. I think one way to do this would be to make the 2 different backends have different "iso"s, then when you curl you can grep the beginning of the data or some other signature to know you hit both. You can just also change it to Eventually with a long timeout to make sure you hit both backends instead of using a for loop 5 times.

Reply:
Ah right... ack let me see if I can improvise this way.

Reply:
I can use the nodeSelector to toggle the advertised routes, so I can first advertise the node on which the backends are present and then I can change the label selector and advertise the node on which the backends are not present and it should still work well. Let's give that a try..

Reply:
oh, cool!

Reply:
OK, this is done! It was interesting to learn about https://metallb.universe.tf/configuration/_advanced_bgp_configuration/#announcing-the-service-from-a-subset-of-nodes -> used that to control which node is being hit.